v2/reader.go

```diff
@@ -13,10 +14,13 @@ import (
 
 // Reader represents a reader of CARv2.
 type Reader struct {
-	Header Header
-	r      io.ReaderAt
-	roots  []cid.Cid
-	closer io.Closer
+	Header      Header
+	Version     uint64
+	r           io.ReaderAt
+	roots       []cid.Cid
+	ropts       ReadOptions
+	carV1Reader *carv1.CarReader
```

Comment:
document that this is only used for Next? for example, it's not obvious that it can't be reused when the user asks for a carv1 reader - because then the offsets would be shared too.

Reply:
Will add extra comments here thanks for pointing this out. I have already added docs to the [`Next`](https://github.com/ipld/go-car/blob/78dd9e9e920e21b62cdd5844c4fdc10478911653/v2/reader.go#L133) function pointing this out.